internal/backends/python/python.go

```diff
@@ -769,20 +761,39 @@ func makePythonUvBackend() api.LanguageBackend {
 
 		return pkgs
 	}
-	addExtraIndexes := func(pkgName string) {
-		extraIndexes, ok := extraIndexMap[pkgName]
-		if ok {
-			uvIndex := os.Getenv("UV_INDEX")
-
-			for _, index := range extraIndexes {
-				if strings.HasPrefix(runtime.GOOS, index.os) {
-					uvIndex = index.url + " " + uvIndex
-				}
-			}
+	addTorchOverride := func() {
+		// check if the override is already added
+		b, err := os.ReadFile("pyproject.toml")
+		if err != nil {
+			return
+		}
+		if strings.Contains(string(b), "https://download.pytorch.org/whl/cpu") {
+			return
+		}
 
-			os.Setenv("UV_INDEX", uvIndex)
+		pyprojectFile, err := os.OpenFile("pyproject.toml", os.O_APPEND|os.O_WRONLY, 0600)
+		if err != nil {
+			return
+		}
+		defer pyprojectFile.Close()
+
+		torchCPUOverride := `
+[tool.uv.sources]
```

Comment:
so the reason this is particularly unpleasant is that TOML doesn't permit duplicate table keys. `[[tool.uv.index]]` might be fine, since it's an array, but it may need to be colocated. `[tool.uv.sources]` is absolutely not OK, and will error if the sources are already defined and we add a duplicate block

Reply:
I see, maybe can we just deserialize it like in `readPyproject`, add the torch overrides, and then serialize it to `pyproject.toml` then?

Reply:
the problem _there_ is that we can't deserialize unless we know the full structure, and then even if we put it back we'll nuke comments and formatting

Reply:
I was thinking we could use something like https://github.com/ceejbot/tomato in order to make this happen, but I haven't actually packaged it into the available runtime